Assets/Scripts/Player/BasePlayerMovement.cs

```diff
@@ -0,0 +1,30 @@
+using UnityEngine;
+
+namespace Player
+{
+	public class BasePlayerMovement : MonoBehaviour
+	{
+		[SerializeField] private float movementSpeed;
+		[SerializeField] private float rotationSpeed;
+
+		private Rigidbody _rb;
+		private Quaternion _direction;
+
+		private void Awake()
+		{
+			_rb = GetComponent<Rigidbody>();
+			_direction = _rb.rotation;
+		}
+
+
+		private void FixedUpdate()
+		{
+			float horizontal = Input.GetAxis("Horizontal");
```

Comment:
Wouldn't it be better to put this in the normal Update loop?

Reply:
Physics belong in the FixedUpdate loop, and getting Input in the normal frame Update loop would have no effect, if you can only apply it once every FixedUpdate iteration.